platform/glfw/glfw_view.cpp

```diff
@@ -933,3 +954,63 @@ void GLFWView::toggleCustomSource() {
                              mbgl::style::VisibilityType::None : mbgl::style::VisibilityType::Visible);
     }
 }
+
+void GLFWView::toggleLocationComponentLayer() {
+    if (map->getStyle().getLayer("puck") == nullptr) {
+        auto puckLayer = std::make_unique<mbgl::style::LocationComponentLayer>("puck");
+
+        puckLayer->setBearing(0);
+        puckLayer->setLocation(LatLng{35.683389, 139.76525});
+        puckLayer->setAccuracyRadius(50);
+        puckLayer->setAccuracyRadiusColor(mbgl::Color{0.0, 1.0, 0.0, 0.05});
+        puckLayer->setAccuracyRadiusBorderColor(mbgl::Color{0.0, 1.0, 0.0, 0.25});
+        puckLayer->setTopImageSize(24);
+        puckLayer->setBearingImageSize(72);
+        puckLayer->setShadowImageSize(96);
+        puckLayer->setImageTiltDisplacement(8.0f); // set to 0 for a "flat" puck
+        puckLayer->setPerspectiveCompensation(0.9);
+
+        map->getStyle().addImage(std::make_unique<mbgl::style::Image>(
+            "puck.png", mbgl::decodeImage(mbgl::util::read_file(mbglPuckAssetsPath + "puck.png")), 1.0));
+
+        map->getStyle().addImage(std::make_unique<mbgl::style::Image>(
+            "puck_shadow.png", mbgl::decodeImage(mbgl::util::read_file(mbglPuckAssetsPath + "puck_shadow.png")), 1.0));
+
+        map->getStyle().addImage(std::make_unique<mbgl::style::Image>(
+            "puck_hat.png", mbgl::decodeImage(mbgl::util::read_file(mbglPuckAssetsPath + "puck_hat.png")), 1.0));
+
+        puckLayer->setBearingImage(expression::Image("puck.png"));
+        puckLayer->setShadowImage(expression::Image("puck_shadow.png"));
+        puckLayer->setTopImage(expression::Image("puck_hat.png"));
+
+        puck = puckLayer.get();
+        map->getStyle().addLayer(std::move(puckLayer));
+    } else {
+        bool visible = !bool(int(puck->getVisibility()));
+        if (visible) {
+            if (!puckFollowsCameraCenter) {
+                mbgl::LatLng mapCenter = map->getCameraOptions().center.value();
+                puck->setLocation(mapCenter);
+                puckFollowsCameraCenter = true;
+            } else {
+                puckFollowsCameraCenter = false;
+                puck->setVisibility(mbgl::style::VisibilityType(mbgl::style::VisibilityType::None));
+            }
+        } else {
+            puck->setLocation(LatLng{35.683389, 139.76525});
+            puck->setVisibility(mbgl::style::VisibilityType(mbgl::style::VisibilityType::Visible));
+            puckFollowsCameraCenter = false;
+        }
+    }
+}
+
+using Nanoseconds = std::chrono::nanoseconds;
+
+void GLFWView::onWillStartRenderingFrame() {
+    uint64_t ns = mbgl::Clock::now().time_since_epoch().count();
+    const float bearing = float(ns % 2000000000) / 2000000000.0 * 360.0; // testing only
```

Comment:
could you give more details on this^?

Reply:
a way to make the location component rotate continuously